src/components/PsAccounts.vue

```diff
@@ -1,130 +1,145 @@
         <BaseOverlay
+          v-if="hasSlotContent($slots.default)"
           class="acc-mt-4"
-          :show="!hasAllShopsLinked">
+          :show="!!shopsToLink.length"
+        >
           <slot />
           <slot name="body" />
         </BaseOverlay>
-        <div class="acc-mt-4">
+        <div
+          v-if="hasSlotContent($slots.customBody)"
+          class="acc-mt-4"
+        >
           <slot name="customBody" />
         </div>
       </template>
     </template>
   </div>
 </template>
 
-<script lang="ts">
-import {
-  computed, defineComponent, onMounted, PropType, ref,
-} from 'vue-demi';
-import {Context} from '@/types/context';
-import AccountPanel from '@/components/panel/AccountPanel.vue';
-import BaseAlert from '@/components/alert/BaseAlert.vue';
-import BaseOverlay from '@/components/BaseOverlay.vue';
-import PsAccountComponentAlertDisplay from '@/containers/PsAccountComponentAlertDisplay.vue';
-import usePSAccountsContext from '@/composables/usePSAccountsContext';
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import { Context, Shop, ShopContext } from '@/types/context';
+import { contextSchema } from '@/lib/ContextValidator';
 import '@/assets/css/index.css';
-
-/**
+import { hasSlotContent } from '@/lib/utils';
+  /**
    * `PsAccounts` will automate pre-requisites checks and will call sub-components directly
    * to ensure each functional case is covered for you. You can use the default slots
    * that will be disabled if the user account is not well linked (you should put your
    * module configuration panel here)
    */
-export default defineComponent({
-  name: 'PsAccounts',
-  components: {
-    AccountPanel,
-    BaseAlert,
-    BaseOverlay,
-    PsAccountComponentAlertDisplay,
-  },
-  props: {
+  interface PsAccountsProps {
     /**
-         * The whole context object given
-         * [by ps\_accounts module presenter function](https://github.com/PrestaShopCorp/prestashop-accounts-installer#register-as-a-service-in-your-psx-container-recommended).
-         * If left empty (by default), the context will be retrieved from JS global
-         * var window.contextPsAccounts automatically.
-         */
-    context: {
-      type: Object as PropType<Context>,
-      required: false,
-      default: () => window.contextPsAccounts || {},
-    },
-  },
-  setup(props) {
-    const {
-      context,
-      setContext,
-      shopsInContext,
-    } = usePSAccountsContext();
-    const hasError = ref(false);
+    * The whole context object given
+    * [by ps\_accounts module presenter function](https://github.com/PrestaShopCorp/prestashop-accounts-installer#register-as-a-service-in-your-psx-container-recommended).
+    * If left empty (by default), the context will be retrieved from JS global
+    * var window.contextPsAccounts automatically.
+    */
+    context?: Context;
+  }
+const props = withDefaults(defineProps<PsAccountsProps>(), {
+  context: () => window.contextPsAccounts || {}
```

Comment:
Pinia cannot be useful for the state ? VueX throwed error but if Pinia can work, it can be useful to put all getter/filter for context in one place (or many, if we split the context in multiple entity). For exemple, methods below like _shops_ or _shopsInContext_ and so many in different components could be pinia getters

Reply:
Yes totally